libc/dirent/closedir.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * test/test_strftime.c
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Arkadiusz Kozlowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <time.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <pthread.h>
+#include <signal.h>
+
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <string.h>
+
+#include <unity_fixture.h>
+
+#include "common.h"
+
+#define MAIN_DIR          "test_closedir"
+#define INTERRUPT_DIR_NUM 256
+
+
+TEST_GROUP(dirent_closedir);
+
+
+TEST_SETUP(dirent_closedir)
+{
+	mkdir(MAIN_DIR, 0777);
+	mkdir(MAIN_DIR "/dir1", 0777);
+}
+
+
+TEST_TEAR_DOWN(dirent_closedir)
+{
+	errno = 0;
+	rmdir(MAIN_DIR "/dir1");
+	rmdir(MAIN_DIR);
+}
+
+
+TEST(dirent_closedir, closing_empty_dir)
+{
+	errno = 0;
+	DIR *dp = opendir(MAIN_DIR "/dir1");
+
+	TEST_ASSERT_NOT_NULL(dp);
+	TEST_ASSERT_EQUAL(0, closedir(dp));
+}
+
+
+TEST(dirent_closedir, closing_non_empty_dir)
+{
+	errno = 0;
+	DIR *dp = opendir(MAIN_DIR);
+
+	TEST_ASSERT_NOT_NULL(dp);
+	TEST_ASSERT_EQUAL(0, closedir(dp));
```

Comment:
I think we should use asserts with explicit type everywhere in this case TEST_ASSERT_EQUAL_INT